vulture_os/toolkit/api_parser/csc_domainmanager/csc_domainmanager.py

```diff
@@ -0,0 +1,201 @@
+            raise CscDomainManagerAPIError(e)
+
+
+    def _format_log(self, log:dict) -> dict:
+        """
+        Format input log
+        """
+        try:
+            timestamp = datetime.strptime(log['event'].get('date'),"%Y-%m-%dT%H:%M:%SZ").timestamp()
+            if self.last_log_time < timestamp:
+                self.last_log_time = timestamp
+        except Exception as e:
+            logger.error(f"[{__parser__}]:format_log: Exception while updating last_log_time logs -- {e}", extra={'frontend': str(self.frontend)})
+            raise CscDomainManagerAPIError(e)
+        try:
+            return dumps(log)
+        except Exception as e:
+            logger.error(f"[{__parser__}]:format_log: Bad json formatting -- {e}", extra={'frontend': str(self.frontend)})
+            raise CscDomainManagerAPIError(e)
+
+    def test(self):
+        try:
+            logs = self.get_logs(since=(timezone.now() - timedelta(hours=12)))
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            logger.exception(f"{[__parser__]}:test: {str(e)}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def execute(self):
+        """
+        Central function to fetch logs
+        """
+        try:
+            page = 1
+            logs = self.get_logs(page=page, since=self.last_api_call or timezone.now() - timedelta(hours=1))
+            self.update_lock()
+
+            if not logs['links'].get('next') and logs['events']:
+                self.write_to_file([self._format_log(event) for event in logs['events']])
+                self.update_lock()
+                if self.last_log_time != 0:
+                    self.last_api_call = datetime.fromtimestamp(self.last_log_time)
+                    self.frontend.last_api_call = self.last_api_call
+
+            elif logs['links'].get('next') and logs['events']:
+                self.write_to_file([self._format_log(event) for event in logs['events']])
+                self.update_lock()
+                page = page + 1
+                self.session = None
+                lock = True
+                while lock:
+                    logs = self.get_logs(page=page, since=self.last_api_call or timezone.now() - timedelta(hours=1))
+                    self.update_lock()
+                    self.write_to_file([self._format_log(event) for event in logs['events']])
+                    self.update_lock()
+                    page = page + 1
+                    self.session = None
+
+                    if logs['events'] and not logs['links'].get('next'):
+                        lock = False
+
+                if self.last_log_time != 0:
+                    self.last_api_call = datetime.fromtimestamp(self.last_log_time)
+                    self.frontend.last_api_call = self.last_api_call
```

Comment:
lots of complexity and duplicated code, consider simplifying/factorizing
- logs['events'] can be taken and written as soon as it exists and is not null, without any `if...else` conditions
- the `logs['links'].get('next')` can be used as a variable that replaces the current `lock` variable
  - as long as `logs['links'].get('next')` is true, the while continues: 
    - increment `page`
    - get logs
    - write logs
    - update_lock

Also: 
- don't forget to limit the queried period with a limited `to` (if possible with the API, see previous comment)
- check self.evt_stop in while conditions to stop parser if required